argparse_dataclass.py

```diff
@@ -338,6 +341,11 @@ def _add_dataclass_options(
 
         if field.type is bool:
             _handle_bool_type(field, args, kwargs)
+        elif get_origin(field.type) is typing.Union:
+            # Optional[X] is equivalent to Union[X, None].
+            f_args = get_args(field.type)
+            if len(f_args) == 2 and f_args[1] is type(None):
```

Comment:
This seems a bit fragile to me (e.g., what if we later add support for other union types?). Wouldn't it be sufficient to just check if `None` is any one of the types in the `Union`?

Reply:
I am fine with updating it to be more general. The documentation for [Optional ](https://docs.python.org/3.8/library/typing.html?highlight=get_args#typing.Optional) states it is equivalent to ```Union[X, None]```, so I thought it was safe to assume the order would not change.